backend/api/pipeline.proto

```diff
@@ -75,6 +75,13 @@ service PipelineService {
     };
   }
 
+    // Finds a specific pipeline by Name (within a namespace).
+    rpc GetPipelineByName(GetPipelineByNameRequest) returns (Pipeline) {
+      option (google.api.http) = {
+        get: "/apis/v1beta1/pipelines/name/{name}"
```

Comment:
Thank you for working on this!
I search for the URL convention and found https://stackoverflow.com/questions/20381976/rest-api-design-getting-a-resource-through-rest-with-different-parameters-but.

There are some different opinions on what this type of an endpoint should look like. I'm curious about your rationale when choosing this.

Reply:
The URL I suggest in the PR is the one I thought was the most obvious choice at the begging of the implementation, but I was more focused on the name itself, not thinking about the namespace.  And the result cURL looks like this: 

```
localhost:8888/apis/v1beta1/pipelines/name/<name>?resource_reference_key.id=kubeflow&resource_reference_key.type=NAMESPACE
```

Now you make me think twice, and probably better URL would be:  

``` get: "/apis/v1beta1/pipelines/name/{name}/namespace/{namespace}"```
WDYT?

Reply:
I agree with that direction, but probably `/apis/v1beta1/namespace/{namespace}/pipelines/{name}` feels closer to REST convention?

Reply:
Although I totally agree that REST convention matches common sense of API definition, I would like to bring up two points:

1. The current usage of namespace filter is through resource_reference in KFP APIs: You can search `namespace` in https://www.kubeflow.org/docs/components/pipelines/reference/api/kubeflow-pipeline-api-spec/.
2. Namespace field is required in full-fledged Kubeflow deployment (with multi-user mode), and not required in standalone KFP deployment (use same namespace as KFP workload). Embedding `namespaces/{namespace}` to API path will forces standalone KFP API request to fill in namespace field.

Reply:
Happy New 2022 Year! 

@zijianjoy As I get, your suggestion matches my current implementation - using resource_reference(type and id) to filter for namespace. 
Now, taking into consideration @zijianjoy comment ... Because of consistency point of view and namespace not required in standalone mode, I guess it would be better to stick with the resource_reference filtering.

WDYT @Bobgy, @zijianjoy ?